packages/assets-controllers/src/Standards/NftStandards/ERC721/ERC721Standard.ts

```diff
@@ -176,28 +177,23 @@ export class ERC721Standard {
       throw new Error("This isn't a valid ERC721 contract");
     }
 
-    let tokenURI, image, symbol, name;
-
-    // TODO upgrade to use Promise.allSettled for name/symbol when we can refactor to use es2020 in tsconfig
-    try {
-      symbol = await this.getAssetSymbol(address);
-    } catch {
-      // ignore
-    }
-
-    try {
-      name = await this.getAssetName(address);
-    } catch {
-      // ignore
-    }
-
-    if (tokenId) {
+    const [symbol, name, tokenURI] = await Promise.all([
```

Comment:
and here

Reply:
It's true that `Promise.allSettled` could be used here.  In this case `safelyExecute` wraps all the promises, which returns undefined if an error occurs.  It was a little cleaner than the `Promise.allSettled` version, where the compiler requires you to check the status before accessing the value.